### ISSUE TITLE
Error: Invalid ios settings: $(MARKETING_VERSION)

### ISSUE THREAD

**Issue by cviebrock** [HausennTechnologies/capacitor-set-version#188]
In my Ionic project, I have this in `package.json`:
```json
"scripts": {
  "capacitor:sync:before": "capacitor-set-version ."
}
```
(the dot is required, otherwise it complains about a missing arg: `dir  Capacitor project root directory`, but that's not my issue).

This seems to work fine for Android, but when it gets to iOS, I get the error:
```
›   Error: Invalid ios settings: $(MARKETING_VERSION)
```
I'm running Xcode: 13.1, and some Googling seems to suggest this was a change with Xcode 11?

Have I done something wrong, or is this a bug with the package?

**Comment by DKrepsky** [comment]
Indeed, this is a change to Xcode. All version information was moved from the Info.plist file to the project settings file.

A fix for this should arrive next week.

Regards

**Comment by chrisDupuis** [comment]
in utils.ts

import * as xcode from 'xcode';

// export const iosFile = '/ios/App/App/Info.plist';
export const iosFile = '/ios/App/App.xcodeproj/project.pbxproj';



  } catch (error) {
    return null;
  }
}

export function setAndroidVersion(options: { dir: string; version: string }): void {
  const file = fs.readFileSync(options.dir + androidFile, 'utf-8');
  const result = file.replace(/(.*(?:versionName).*)/g, `        versionName "${options.version}"`);
  fs.writeFileSync(options.dir + androidFile, result, 'utf-8');
}

export function getAndroidBuild(options: { dir: string }): number | null {
  try {
    const file = fs.readFileSync(options.dir + androidFile, 'utf-8');
    const buid = file.split('versionCode ')[1].split('\n')[0];
    return parseInt(buid);
  } catch (error) {
    return null;
  }
}

export function setAndroidBuild(options: { dir: string; build: number }): void {
  const file = fs.readFileSync(options.dir + androidFile, 'utf-8');
  const result = file.replace(/(.*(?:versionCode).*)/g, `        versionCode ${options.build}`);
  fs.writeFileSync(options.dir + androidFile, result, 'utf-8');
}

export async function getIOSVersion(options: { dir: string }): Promise<string | null> {
  try {
    // const file = fs.readFileSync(options.dir + iosFile, 'utf-8');
    // const parsed = plist.parse(file) as any;
    // return parsed.CFBundleShortVersionString;

    const myProj = xcode.project(options.dir + iosFile);
    await myProj.parseSync();
    return myProj.getBuildProperty('MARKETING_VERSION', 'Release');
  } catch (error) {
    return null;
  }
}

export async function setIOSVersion(options: { dir: string; version: string }): Promise<void> {
  // const file = fs.readFileSync(options.dir + iosFile, 'utf-8');
  // const parsed = plist.parse(file) as any;
  // parsed.CFBundleShortVersionString = options.version;
  // const result = plist.build(parsed);
  // fs.writeFileSync(options.dir + iosFile, result, 'utf-8');

  const myProj = xcode.project(options.dir + iosFile);
  await myProj.parseSync();
  myProj.updateBuildProperty('MARKETING_VERSION', options.version, 'Release');
  myProj.updateBuildProperty('MARKETING_VERSION', options.version, 'Debug');
  await fs.writeFileSync(options.dir + iosFile, myProj.writeSync());
}

export async function getIOSBuild(options: { dir: string }):Promise<number | null> {
  try {
  //   const file = fs.readFileSync(options.dir + iosFile, 'utf-8');
  //   const parsed = plist.parse(file) as any;
  //   return parseInt(parsed.CFBundleVersion);
    const myProj = xcode.project(options.dir + iosFile);
    await myProj.parseSync();
    return myProj.getBuildProperty('CURRENT_PROJECT_VERSION', 'Release');
  } catch (error) {
    return null;
  }
  return null;
}

export async function setIOSBuild(options: { dir: string; build: number }): Promise<void> {
  // const file = fs.readFileSync(options.dir + iosFile, 'utf-8');
  // const parsed = plist.parse(file) as any;
  // parsed.CFBundleVersion = options.build.toString();
  // const result = plist.build(parsed);
  // fs.writeFileSync(options.dir + iosFile, result, 'utf-8');

  const myProj = xcode.project(options.dir + iosFile);
  await myProj.parseSync();
  myProj.updateBuildProperty('CURRENT_PROJECT_VERSION', options.build, 'Release');
  myProj.updateBuildProperty('CURRENT_PROJECT_VERSION', options.build, 'Debug');
  await fs.writeFileSync(options.dir + iosFile, myProj.writeSync());
}

**Comment by DKrepsky** [comment]
Thanks you @chrisDupuis 

We worked around this by disabling version management on xcode, but you solution is better.

I'll open a PR for this.

**Comment by dylanvdmerwe** [comment]
Any news regarding a fix for this issue?
` Error: Invalid ios settings: $(MARKETING_VERSION)`


   "capacitor-set-version": "^1.3.35"

**Comment by DKrepsky** [comment]
Hi,

This should be fixed in v2.x
